connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java

```diff
@@ -79,4 +79,20 @@ public void commit() throws InterruptedException {
      * {@link java.nio.channels.Selector#wakeup() wakeup()} to interrupt any ongoing requests.
      */
     public abstract void stop();
+
+  /**
+   * <p>
+   * Commit an individual {@link SourceRecord} when the callback from the producer client is received.
+   * </p>
+   * <p>
+   * SourceTasks are not required to implement this functionality; Kafka Connect will record offsets
+   * automatically. This hook is provided for systems that also need to store offsets internally
+   * in their own system.
+   * </p>
+   * @param record {@link SourceRecord} that was successfully sent via the producer.
+   * @throws InterruptedException
+   */
+    public void commitRecord(SourceRecord record) throws InterruptedException {
```

Comment:
I left the original commit to allow a developer to do a batch commit if they wanted. My thoughts were that they could build a list of incoming commitRecord() and perform a batch when commit() was called. This would give them more flexibility.